src/java/org/apache/cassandra/index/sai/disk/v1/segment/SegmentRamBuffer.java

```diff
@@ -55,14 +56,17 @@ public long memoryUsed()
         return trie.sizeOnHeap() + postingsAccumulator.heapAllocations();
     }
 
-    public long add(int segmentRowId, byte[] value)
+    public long add(ByteComparable term, int termLength, int segmentRowId)
     {
         final long initialSizeOnHeap = trie.sizeOnHeap();
         final long reducerHeapSize = postingsAccumulator.heapAllocations();
 
         try
         {
-            trie.putRecursive(v -> ByteSource.fixedLength(value), segmentRowId, postingsAccumulator);
+            if (termLength <= MAX_RECURSIVE_TERM_LENGTH)
+                trie.putRecursive(term, segmentRowId, postingsAccumulator);
+            else
+                trie.apply(Trie.singleton(term, segmentRowId), postingsAccumulator);
```

Comment:
Maybe this can be simplified to:
```java
trie.putSingleton(term, segmentRowId, postingsAccumulator, termLength <= MAX_RECURSIVE_TERM_LENGTH);
```
There is a similar usage pattern in `TrieMemoryIndex#addTerm(PrimaryKey, ByteBuffer)`.